zson/marshal_test.go

```diff
@@ -0,0 +1,98 @@
+package zson_test
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/brimsec/zq/zson"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func trim(s string) string {
+	return strings.TrimSpace(s) + "\n"
+}
+
+type Thing interface {
+	Color() string
+}
+
+type Plant struct {
+	MyColor string
+}
+
+func (p *Plant) Color() string { return p.MyColor }
+
+type Animal struct {
+	MyColor string
+}
+
+func (a *Animal) Color() string { return a.MyColor }
+
+func Make(which, color string) Thing {
+	switch which {
+	case "plant":
+		return &Plant{color}
+	case "animal":
+		return &Animal{color}
+	default:
+		return nil
+	}
+}
+
+func TestInterfaceMarshal(t *testing.T) {
+	rose := Make("plant", "red")
+	expectedRose := `{MyColor: "red"} (=Plant)`
+	flamingo := Make("animal", "pink")
```

Comment:
```suggestion

func TestInterfaceMarshal(t *testing.T) {
	rose := Thing(&Plant{"red"})
	expectedRose := `{MyColor: "red"} (=Plant)`
	flamingo := Thing(&Animal{"pink"})
```

Reply:
This came from my article draft to be pedantic.  I will simplify the test here.